metric_config_parser/tests/test_config.py

```diff
@@ -503,3 +503,157 @@ def test_invalid_metric_level(self):
+
+            [metrics.'test_me*']
+            category = "test_overwrite_second"
+            """
+        )
+        definition = DefinitionConfig(
+            slug="firefox_desktop",
+            platform="firefox_desktop",
+            spec=AnalysisSpec.from_dict(toml.loads(config_str)),
+            last_modified=datetime.datetime.now(),
+        )
+        config_collection_2 = ConfigCollection(
+            configs=[], outcomes=[], defaults=[], definitions=[definition]
+        )
+
+        config_collection_1.get_metric_definition(
+            "test_metric", "firefox_desktop"
+        ).data_source is None
+        assert (
+            config_collection_1.get_data_source_definition(
+                "baseline", "firefox_desktop"
+            ).experiments_column_type
+            == "simple"
+        )
+
+        config_collection_1.merge(config_collection_2)
+        assert (
+            config_collection_1.get_data_source_definition("baseline", "firefox_desktop").name
+            == "baseline"
+        )
+        assert (
+            config_collection_1.get_data_source_definition(
+                "baseline", "firefox_desktop"
+            ).experiments_column_type
+            == "none"
+        )
+        assert config_collection_1.get_data_source_definition("*", "firefox_desktop") is None
+        assert (
+            config_collection_1.get_metric_definition(
+                "test_metric", "firefox_desktop"
+            ).data_source.name
+            == "baseline"
+        )
+        assert (
+            "bootstrap_mean"
+            in config_collection_1.get_metric_definition(
+                "test_metric", "firefox_desktop"
+            ).statistics
+        )
+        assert (
+            config_collection_1.get_metric_definition("test_metric", "firefox_desktop").category
+            == "test_overwrite_second"
+        )
+
+    def test_merge_with_wildcards_invalid(self):
+        config_str = dedent(
+            """
+            [data_sources.baseline]
+            from_expression = "mozdata.search.baseline"
+            experiments_column_type = "simple"
+            """
+        )
+
+        definition = DefinitionConfig(
+            slug="firefox_desktop",
+            platform="firefox_desktop",
+            spec=AnalysisSpec.from_dict(toml.loads(config_str)),
+            last_modified=datetime.datetime.now(),
+        )
+        config_collection_1 = ConfigCollection(
+            configs=[], outcomes=[], defaults=[], definitions=[definition]
+        )
+
+        config_str = dedent(
+            """
+            [data_sources.'invalid_*']
+            experiments_column_type = "none"
+            """
+        )
+        definition = DefinitionConfig(
+            slug="firefox_desktop",
+            platform="firefox_desktop",
+            spec=AnalysisSpec.from_dict(toml.loads(config_str)),
+            last_modified=datetime.datetime.now(),
+        )
```

Comment:
Shouldn't this raise a `ValueError`?

Reply:
The `ValueError` is only raised when calling `.resolve()` on the `DataSourceDefinition`. The method isn't called in this test though, `get_data_source_definition()` etc are just returning the `DataSourceDefinition`

Reply:
OK, that makes sense, but `test_invalid_wildcard_in_data_source` doesn't seem to call `resolve` either?

Reply:
Right, I added a comment to the test.
`test_invalid_wildcard_in_data_source` checks that when using a wildcard in an identifier it needs to be wrapped in `'`. So it expects `[data_sources.'baseline_*']` instead of `[data_sources.baseline_*]`